pkg/server/server.go

```diff
@@ -193,10 +193,6 @@ func (s *Server) Run(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
-	servingOpts := s.options.GenericControlPlane.SecureServing
-	if err := s.options.AdminAuthentication.WriteKubeConfig(genericConfig, newTokenOrEmpty, tokenHash, s.options.GenericControlPlane.GenericServerRunOptions.ExternalHost, servingOpts.BindPort); err != nil {
```

Comment:
Curious: why did this move down toward the bottom?

Reply:
external address defaulting happens behind the `apisConfig.Complete()`. I think he's making sure that's done before writing the kubeconfig.

Reply:
we need to call `Complete` on `genericConfig` so that the `ExternalAddress` gets [calculated](https://github.com/kcp-dev/kubernetes/blob/10415751c5a9f503df2dca997114a2d4de855340/staging/src/k8s.io/apiserver/pkg/server/config.go#L482-L483).

Reply:
Thanks!